examples/common/config.yaml

```diff
@@ -3,38 +3,53 @@
 # https://www.envoyproxy.io/docs/envoy/latest/configuration/configuration
 static_resources:
   listeners:
-    - address:
-        socket_address: {address: 0.0.0.0, port_value: 9001, protocol: TCP}
-      filter_chains:
-        - filters:
-            - name: envoy.http_connection_manager
-              config:
-                stat_prefix: client
-                route_config:
-                  virtual_hosts:
-                    - name: all
-                      domains: ["*"]
-                      routes:
-                        - match: {prefix: "/"}
-                          route: {cluster: hello_world_api}
-                http_filters:
-                  - name: envoy.router
+  - name: listener_0
+    address:
+      socket_address:
+        protocol: TCP
+        address: 0.0.0.0
+        port_value: 9001
+    filter_chains:
+    - filters:
+      - name: envoy.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
+          stat_prefix: ingress_http
+          http_protocol_options:
+            allow_absolute_url: true
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: local_service
+              domains: ["*"]
+              routes:
+              - match:
+                  prefix: "/"
+                route:
+                  cluster: dynamic_forward_proxy_cluster
+          http_filters:
+          - name: envoy.filters.http.dynamic_forward_proxy
+            config:
+              dns_cache_config:
+                name: dynamic_forward_proxy_cache_config
+                dns_lookup_family: V4_ONLY
+          - name: envoy.router
   clusters:
-    - name: hello_world_api
-      connect_timeout: 30s
-      dns_lookup_family: V4_ONLY
-      dns_refresh_rate: 60s
-      lb_policy: ROUND_ROBIN
-      load_assignment:
-        cluster_name: hello_world_api
-        endpoints:
-          - lb_endpoints:
-              - endpoint:
-                  address:
-                    socket_address: {address: s3.amazonaws.com, port_value: 443}
-      tls_context:
-        sni: s3.amazonaws.com
-      type: LOGICAL_DNS
+  - name: dynamic_forward_proxy_cluster
+    connect_timeout: 30s
+    dns_refresh_rate: 60s
+    lb_policy: CLUSTER_PROVIDED
+    cluster_type:
+      name: envoy.clusters.dynamic_forward_proxy
+      typed_config:
+        "@type": type.googleapis.com/envoy.config.cluster.dynamic_forward_proxy.v2alpha.ClusterConfig
+        dns_cache_config:
+          name: dynamic_forward_proxy_cache_config
+          dns_lookup_family: V4_ONLY
+    # tls_context:
+    #   common_tls_context:
+    #     validation_context:
+    #       trusted_ca: {filename: /etc/ssl/certs/ca-certificates.crt}
```

Comment:
We definitely need/want this. Do we know what the correct paths to load the platform trusted CA files are? Or is it more difficult than that on mobile?

Reply:
Totally agree. We're planning on merging with this temporarily disabled, as we'll likely need to compile our own with the library. Tracking here, and linked within the config: https://github.com/lyft/envoy-mobile/issues/322

Reply:
It's a little scary to merge this as is, in case anyone's actually playing around with this library. Do you want to add logic to manually reject non-https traffic for now?

Reply:
Talked offline - opted to leave a `WARNING!` here since this is in `examples/` and we are still in demo stage.